backends/firrtl/firrtl.cc

```diff
@@ -42,6 +43,23 @@ static const FDirection FD_OUT = 0x2;
 static const FDirection FD_INOUT = 0x3;
 static const int FIRRTL_MAX_DSH_WIDTH_ERROR = 20; // For historic reasons, this is actually one greater than the maximum allowed shift width
 
+std::string getFileinfo(const RTLIL::AttrObject *design_entity)
+{
+	std::string src(design_entity->get_src_attribute());
+
+	std::ostringstream fileinfo;
+	if (!src.empty()) {
+		fileinfo << "@[" << src << "]";
+	}
+
+	// Remove quotes from src attribute as firrtl automatically escapes and
+	// double-quotes them.
+	std::string fileinfo_str(fileinfo.str());
```

Comment:
```suggestion
	std::string fileinfo_str = "@[" + src + "]";
```

Reply:
That's indeed way simpler :)
I'll update it.